improver/metadata/utilities.py

```diff
@@ -44,42 +50,41 @@
 
 
 def create_new_diagnostic_cube(
-    name,
-    units,
-    template_cube,
-    mandatory_attributes,
-    optional_attributes=None,
-    data=None,
-    dtype=np.float32,
-):
+    name: str,
+    units: Union[Unit, str],
+    template_cube: Cube,
+    mandatory_attributes: Union[Dict[str, str], LimitedAttributeDict],
+    optional_attributes: Optional[Any] = None,
+    data: Union[MaskedArray, ndarray] = None,
+    dtype: type = np.float32,
```

Comment:
It's ok to use `type` here. But instead I'll put `Type` which is imported from `typing`. In Python 3.9+ it's possible to use things like `list[str]` instead of `List[str]`. I think using `type` instead of `Type` is similar.